python/taichi/lang/__init__.py

```diff
@@ -521,7 +517,10 @@ def init(arch=None,
     # Check version for users every 7 days if not disabled by users.
     skip = os.environ.get("TI_SKIP_VERSION_CHECK")
     if skip != 'ON':
-        try_check_version()
+        # We don't join this thread because we do not wish to block the user.
+        check_version_thread = threading.Thread(target=try_check_version,
```

Comment:
Just to double check: have you tried locally that just raise in this thread and see if it crash the main process?

Reply:
I tried crash the thread and not catch error. The sub thread will crash and a bunch of output will be printed, but the main thread can run as normal.

Reply:
Yeah, the exceptions raised in sub thread can't fail the main thread. But I am not sure if `sys.exit` would work

Reply:
Can you elaborate about sys.exit please?

Reply:
I may have understood your need, if you don't want it to fail the main program, then subthread is OK.

Reply:
Yes, the main program is the last thing we wish to fail.